numpy/core/tests/test_scalarbuffer.py

```diff
@@ -0,0 +1,107 @@
+"""
+Test scalar buffer interface adheres to PEP 3118
+"""
+import sys
+import numpy as np
+from numpy.testing import run_module_suite, assert_, dec
+
+# types
+scalars = [
+    np.bool_, np.bool8, np.byte, np.short, np.intc, np.int_,
+    np.longlong, np.intp, np.int8, np.int16, np.int32, np.int64,
+    np.ubyte, np.ushort, np.uintc, np.uint, np.ulonglong,
+    np.uintp, np.uint8, np.uint16, np.uint32, np.uint64,
+    np.half, np.single, np.double, np.float_, np.longfloat,
+    np.float16, np.float32, np.float64, np.csingle, np.complex_,
+    np.clongfloat, np.complex64, np.complex128,
+]
+
+scalars_set_code = [
+    (np.bool_, '?'),
+    (np.byte, 'b'),
+    (np.short, 'h'),
+    (np.intc, 'i'),
+    (np.int_, 'l'),
+    (np.longlong, 'q'),
+    (np.ubyte, 'B'),
+    (np.ushort, 'H'),
+    (np.uintc, 'I'),
+    (np.uint, 'L'),
+    (np.ulonglong, 'Q'),
+    (np.half, 'e'),
+    (np.single, 'f'),
+    (np.double, 'd'),
+    (np.float_, 'd'),
+    (np.longfloat, 'g'),
+    (np.csingle, 'Zf'),
+    (np.complex_, 'Zd'),
+    (np.clongfloat, 'Zg'),
+
+    # TODO: 'p', 'l', or 'q'?
+    # (np.intp, 'p'),
+    # (np.uintp, 'P'),
+]
+
+scalars_set_size = [
+    (np.bool8, 1),
+    (np.int8, 1),
+    (np.int16, 2),
+    (np.int32, 4),
+    (np.int64, 8),
+    (np.uint8, 1),
+    (np.uint16, 2),
+    (np.uint32, 4),
+    (np.uint64, 8),
+    (np.float16, 2),
+    (np.float32, 4),
+    (np.float64, 8),
+    (np.complex64, 8),
+    (np.complex128, 16),
```

Comment:
I think all of these would be better as `np.dtype(type).itemsize`, to avoid duplication. You could then use `longdouble` instead of `float128` below, which is always defined.

Reply:
would `type().itemsize` be better than `np.dtype(type).itemsize`? Not sure if there is an appreciable difference.